platforms/substrate/configuration/roles/create/genesis/tasks/main.yaml

```diff
@@ -0,0 +1,73 @@
+##############################################################################################
+#  Copyright Accenture. All Rights Reserved.
+#
+#  SPDX-License-Identifier: Apache-2.0
+##############################################################################################
+
+# This task creates a build directory within the directory that this play is in
+- name: Create build directory if it does not exist
+  file:
+    path: "./roles/create/genesis/build"
+    state: directory
+
+# This task checks if genesis is already in the vault for each org
+- name: Check if genesis is in the vault for each org
+  include_tasks: check_vault.yaml
+  vars:
+    component_ns: "{{ org.name | lower }}-subs"
+    vault: "{{ org.vault }}"
+  loop: "{{ network['organizations'] }}"
+  loop_control:
+    loop_var: org
+
+# This task create the template genesis file and stores it in the build folder
+- name: Create the base genesis file
+  shell: |
+    CONTAINER="{{ network.docker.url }}/{{ network.config.node_image }}:{{ network.version }}"
+    docker pull $CONTAINER
+    BASE_CHAIN="local"
+    GENESIS_DIR="$(pwd)/roles/create/genesis/build"
+    
+    GENESIS=$(docker run --rm -a stdout $CONTAINER build-spec --disable-default-bootnode --chain $BASE_CHAIN)
+    GENESIS=$(echo $GENESIS | jq '.chainType |= "Live"')
+    GENESIS=$(echo $GENESIS | jq '.genesis.runtime.palletAura.authorities |= []')
+    GENESIS=$(echo $GENESIS | jq '.genesis.runtime.palletGrandpa.authorities |= []')
+    echo $GENESIS > $GENESIS_DIR/genesis.json
+  when: generate_genesis is defined and generate_genesis == True
+
+# This task updates the genesis with validator keys
+- name: Update genesis for each validator node
+  include_tasks: nested_main.yaml
+  vars:
+    component_ns: "{{ org.name | lower }}-subs"
+    name: "{{ org.name | lower }}"
+    peers: "{{ org.services.peers }}"
+    peers_name: "{{ org.services.peers.name }}"
+    vault: "{{ org.vault }}"
+  loop: "{{ network['organizations'] }}"
+  loop_control:
+    loop_var: org
+  when: generate_genesis is defined and generate_genesis == True
+
+# This task converts the genesis to raw format 
+- name: Convert the genesis file to raw format
+  shell: |
+    CONTAINER="{{ network.docker.url }}/{{ network.config.node_image }}:{{ network.version }}"
+    GENESIS_DIR="$(pwd)/roles/create/genesis/build"
+
+    GENESIS=$(docker run --rm --mount type=bind,source=$GENESIS_DIR,target=/config $CONTAINER \
+    build-spec --disable-default-bootnode --raw --chain /config/genesis.json)
+    echo $GENESIS > {{ network.config.genesis }}.json 
+    openssl base64 -in {{ network.config.genesis }}.json -out {{ network.config.genesis }}_base64 
+  when: generate_genesis is defined and generate_genesis == True
```

Comment:
This should be removed from here, instead done in a initcontainer or when starting the node. I guess that part is already included in substrate-node chart